File: Integer/FirstBadVersion/FirstBadVersion.cc
class FirstBadVersion {
	public:
		int firstBadVersion(int n) {
			int left = 0;
			int right = n + 1;
			while(left + 1 < right) {
				mid = left + (right - left) / 2;
				if(VersionControll::isBadVersion(mid))
					right = mid;
				else
					left = mid;
				
			}
			return (mid  + 1);
		}
};

File: Integer/FirstBadVersion/FirstBadVersion.py
class FirstBadVersion:
    def firstBadVersion(self, n):
        left, right = 0, n + 1
        while left + 1  < right:
            mid = left + (right -left) / 2
            if VersionControl.isBadVersion(mid):  
                right = mid
            else:
                left = mid
            
        return left + 1